src/meta/processors/schemaMan/CreateEdgeProcessor.cpp

```diff
@@ -11,6 +11,20 @@ namespace meta {
 
 void CreateEdgeProcessor::process(const cpp2::CreateEdgeReq& req) {
     CHECK_SPACE_ID_AND_RETURN(req.get_space_id());
+    {
+        // if there is an tag of the same name
+        folly::SharedMutex::ReadHolder rHolder(LockUtils::tagLock());
```

Comment:
In multithread env, we can not separate this check in two lock, because between read.unlock and write.lock, other thread may insert some same record.

Reply:
Good catch! I have to think about it, I think to use the same lock to do it, but it will handle slower in tag and edge. It is hard to happen when thread A creates tag tagName, thread B create edge tagName at the same timestamp.

Reply:
@wadeliuyi We might as well leave this potential race condition problem to address in the future.

@laura-ding Better to add a TODO here.

Reply:
but "hard to happen" not mean "do not happen", and create tag and edge is not the key path of our program, so performance is not key point?